Feature/193 - Updates for layout of Landing page for NWB Explorer

vidhya-metacell opened the pull request:
Please Note : It consists of changes only for the Splash Page not the File Explorer

1. Followed the new figma designs for landing page
2. Added theme for the above
3. Uses less-to-js-vars for common layout values
4. Custom css is added in Main.less
5. Extra styles for splash page have been removed.
6. Hiding the image on the right side for smaller devices.
7. Page is should work fine for responsive.

tarelli commented:
Fantastic job @vidhya-metacell! Some minor fixes below:

- [ ] Open links in new tab
- [ ] Add hover to experiments buttons (see the style of button one) below which gets darker
<img width="313" alt="Screenshot 2020-09-25 at 14 51 18" src="https://user-images.githubusercontent.com/81127/94275656-1e06b500-ff3f-11ea-832e-aa384001f24a.png">
- [ ] Fix font weights and vertical spacing
<img width="735" alt="Screenshot 2020-09-25 at 14 50 20" src="https://user-images.githubusercontent.com/81127/94275711-3676cf80-ff3f-11ea-8b38-50a0f029c1da.png">
- [ ] Change geppetto loading white page (ask @filippomc how to) as below (just added to Figma):
![NWB Explorer Loading Page](https://user-images.githubusercontent.com/81127/94275923-72aa3000-ff3f-11ea-993a-167bb61f3bdd.png)

tarelli reviewed (changes requested):
See other comment

vidhya-metacell commented:
Completed the following items for FileExplorerPage
(All font sizes for this page are same as NetPyNE font sizes)
(For headings it had 16px and inside content was 0.9rem so kept 15 px)

- AppBar integration -
1) Made similar to NetPyNE topbar
2) Added dialog for About Content
3) Background colour for About Dialog made a little darker as oppose to NetPyNE about us dialog which has white background, as NWB Explorer logo is white coloured.
4) Added link to nwb.org under Help Menu documentation
5) Rest the options that were already available, like Home etc have been moved as AppBar Menu options

- General/Details
1) Made similar to figma designs

- List Viewer
1) Menu integration instead of list controls using gepetto-ui Menu
2) ListMenu, ListMenuContainer, listMenuConfiguration, ListMenuComponent holds the functionality for the menu options
3) Eye icon for viewing the plot and details
4) Hidden all the menu options except for Show Details in case of Image Series
5) Color Picker added from react-color

- Python Console
1) Used the custom.css from the NetPyNE
2) Made changes for font, background color, text color

- PR comments on Splash Page
1) Renamed to Welcome page
2) Button hover state
3) Link open in new tabs
4) Vertical spacing and font fixes

Please Note:
Pending Items
1) Color change for Eye Icon
2) Filter to be implemented later
3) New color picker design to be implemented later
4) NWB loading page

@tarelli @filippomc 
Please review when you get a chance and let me know if I missed something.

vidhya-metacell commented:
- Implemented the Loading screen designs

Pending Items
-Color change for Eye Icon
-Filter to be implemented later
-New color picker design to be implemented later

@filippomc @tarelli

filippomc commented on webapp/components/CustomIconComponent.js:
```diff
@@ -0,0 +1,24 @@
+    title={tooltip}
+    onClick={action}
+  >
+    <RemoveRedEyeIcon />
```
What about parametrizing the icon? Otherwise I'd choose a more specific name for this component

filippomc commented on webapp/components/Dialog.js:
```diff
@@ -0,0 +1,69 @@
+        NWB Explorer is being developed in collaboration with:
+      </Typography>
+      <Link href="http://www.metacell.us" target="_blank">
+        <img style={{ width: 150, padding: "10px" }} src="https://raw.githubusercontent.com/ddelpiano/bucket/master/MetaCellLogoHorizontal300ppi.png"></img>
```
This is coming from netpyne and should be improved also there. The image shouldn't come from outside and the css as well can be improved. "10px padding" is a kind of misuse we are trying to remove from the codebases as we can, unless there is a specific reason to keep things like that.

filippomc commented on webapp/components/Dialog.js:
```diff
@@ -0,0 +1,69 @@
+    </Box>
+    
+    <Box m={1} >
+      <Typography variant="body2" style={{ color: fontColor }}>
```
Shouldn't need to set the font color with style when using typography

filippomc commented on webapp/components/FileUrlSelector.js:
```diff
@@ -31,24 +31,28 @@ export default class FileUrlSelector extends React.Component {
             disabled={this.state.inputValue.length <= 5}
-          >Load NWB file</Button>
-        </div>
+          >LOAD NWB FILE</Button>
```
Text should be lower case, the text-transform is about styling (should be the default for all buttons)

vidhya-metacell replied:
This on has uppercase text in figma, so did the same

filippomc commented on webapp/components/NWBListViewer.js:
```diff
@@ -144,13 +145,13 @@ export default class NWBListViewer extends Component {
       
       <div className='list-summary'>
+        <a title="Plot all timeseries" onClick={this.plotAllInstances}>
+          <RemoveRedEyeIcon />
```
There is another component with the remove eye icon, maybe not used anymore?

vidhya-metacell replied:
Made that parameterized, so not changing here

filippomc commented on webapp/components/appBar/AppBar.js:
```diff
@@ -17,12 +11,12 @@ const styles = theme => ({
     backgroundColor: theme.palette.common.white
   },
-  popper: { paddingRight: "10px" }
+  popper: { paddingRight: "10px" },
```
Use theme.spacing

filippomc commented on webapp/components/configuration/listMenuConfiguration.js:
```diff
@@ -0,0 +1,137 @@
+          backgroundColor: bgDark,
+          borderRadius: 0,
+          color: "#ffffff",
+          fontSize: 14,
```
It would be nice to avoid these kind of custom css and have the menu component using the main theme by default. Surely an improvement to keep in mind on the Geppetto client side.

filippomc commented on webapp/components/configuration/listMenuConfiguration.js:
```diff
@@ -0,0 +1,137 @@
+      {
+        label: "",
+        position: "bottom-start",
+        icon: <i className="fa fa-chevron-down" />,
```
Doesn't work

filippomc reviewed (commented):
Looks very good, need to finish with the mentioned missing color feature. Left a few minor comments to share thoughts about old code parts that we want to improve in the future.

vidhya-metacell commented:
@filippomc 
Added all the below items

- Show eye icon for every type

- Menu option only for image and condition shown

- Hide Add to existing plots menu option if no plots available

- Menu dropdown icon is not visible

- Eye icon color random color

- font weight 700

- Eye icon colour update from color picker selection

- Show Image Series, new menu option, showing only image series not details
-Some PR comments fixes

filippomc reviewed (approved):
Thanks Vidhya, I like what I see!
